components/brave_wallet_ui/common/async/wallet_async_handler.ts

```diff
@@ -176,4 +189,21 @@ handler.on(WalletActions.updateVisibleTokens.getType(), async (store, payload: s
   await refreshWalletInfo(store)
 })
 
+handler.on(WalletActions.selectPortfolioTimeline.getType(), async (store, payload: AssetPriceTimeframe) => {
+  store.dispatch(WalletActions.portfolioTimelineUpdated(payload))
+
+  const apiProxy = await getAPIProxy()
+  const assetPriceController = apiProxy.assetRatioController
+  const state = getWalletState(store)
+  const getTokenPriceHistory = await Promise.all(state.accounts.map(async (account) => {
+    return Promise.all(account.tokens.map(async (token) => {
+      return {
+        token: token,
+        history: await assetPriceController.getPriceHistory(token.asset.symbol.toLowerCase(), state.selectedPortfolioTimeline)
+      }
+    }))
+  }))
```

Comment:
The `getTokenPriceHistory` code looks duplicated from `refreshWalletInfo`. Can we have a dedicated function for `getTokenPriceHistory`?

Reply:
Separated out into it's own function!